package_repo_resource.md

```diff
@@ -0,0 +1,71 @@
+---
+RFC: unassigned
+Author: Thom May <tmay@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Add a resource to manage package repositories
+
+Provide a core resource to allow a user to add a package repository to
+their system.
+
+## Motivation
+
+    As a cookbook author,
+    I want to write cookbooks that rely only on core resources,
+    so that I can minimise my dependencies.
+
+## Specification
+
+Create a new `package_repo` resource, similar to 
+```
+package_repo "foo" do
+  description "A descriptive string"
+  base_url "https://some/url"
+  gpgkey "https://some/other/url"
```

Comment:
I'm all for moving yum_repository and apt_repository into core, but a
generic "package_repo" abstraction feels wrong to me.

The "base_url vs uri" question that came up in the first 5 minutes here is
a perfect example to why.




On Wed, Mar 4, 2015 at 1:10 PM, Dan DeLeo <notifications@github.com> wrote:

> In package_repo_resource.md
> <https://github.com/chef/chef-rfc/pull/98#discussion_r25797488>:
>
> > +their system.
> > +
> > +## Motivation
> > +
> > +    As a cookbook author,
> > +    I want to write cookbooks that rely only on core resources,
> > +    so that I can minimise my dependencies.
> > +
> > +## Specification
> > +
> > +Create a new `package_repo` resource, similar to
> > +```
> > +package_repo "foo" do
> > +  description "A descriptive string"
> > +  base_url "https://some/url"
> > +  gpgkey "https://some/other/url"
>
> @someara <https://github.com/someara> are you referring to the feature in
> general? I'd guess that there's a fair bit of difference between different
> platforms, but generally you get a list of available packages over the
> network and then you can do package-manager install $thing and it works.
> Even if the usage will be quite a bit different between different
> platforms, this is such a common config management task that we should have
> it in core (IMO).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef-rfc/pull/98/files#r25797488>.
>